File: app/src/main/java/com/example/capture/MusicServiceInterface.java
package com.example.capture;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.ServiceConnection;
import android.net.Uri;
import android.os.IBinder;
import android.util.Log;

import com.example.capture.services.MusicService;

import java.util.ArrayList;

public class MusicServiceInterface {
    private ServiceConnection mServiceConnection;
    public MusicService mService;

    public MusicServiceInterface(Context context) {
        mServiceConnection = new ServiceConnection(){
            @Override
            public void onServiceConnected(ComponentName name, IBinder service) {
                mService = ((MusicService.LocalBinder) service).getService();
                Log.d("Interface" , "서비스 연결");
            }

            @Override
            public void onServiceDisconnected(ComponentName name) {
                mServiceConnection = null;
                mService = null;
                Log.d("Interface" , "서비스 연결 해제");
            }
        };
        context.bindService(new Intent(context, MusicService.class)
                .setPackage(context.getPackageName()), mServiceConnection, Context.BIND_AUTO_CREATE);
    }

    public void setmSongList(ArrayList<Uri> songList) {
        if (mService != null) {
            mService.setmSongList(songList);
        }
    }

    public boolean isPlaying() {
        if (mService != null) {
            return mService.isPlaying();
        }
        return false;
    }

    public void play(Uri uri){
        if (mService != null) {
            mService.playMusic(uri);
        }
    }

    public void nextMusic(){
        if (mService != null) {
            mService.nextMusic();
        }
    }

    public void prevMusic(){
        if (mService != null) {
            mService.prevMusic();
        }
    }

    public void clickResumeButton(){
        if (mService != null) {
            mService.clickResumeButton();
        }
    }


    // 사용 안함
//    public AudioAdapter.AudioItem getAudioItem() {
//        if (mService != null) {
//            return mService.getAudioItem();
//        }
//        return null;
//    }

    public void queryAudioItem(int position) {
        if (mService != null) {
            mService.queryAudioItem(position);
        }
    }

    public AudioAdapter.AudioItem mAudioItem() {
        if (mService != null) {
            return mService.mAudioItem;
        }
        return null;
    }

}

File: app/src/main/java/com/example/mediaplayer/MusicServiceInterface.kt
package com.example.mediaplayer

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import android.net.Uri
import android.os.Build
import android.os.IBinder
import android.util.Log
import androidx.annotation.RequiresApi
import com.example.mediaplayer.AudioAdapter.AudioItem
import com.example.mediaplayer.services.MusicService
import com.example.mediaplayer.services.MusicService.LocalBinder
import java.util.*

class MusicServiceInterface constructor(context: Context) {
    private var mServiceConnection: ServiceConnection?
    var mService: MusicService? = null
//    fun setmSongList(songList: ArrayList<Uri>) {
//        if (mService != null) {
//            mService!!.setmSongList(songList)
//        }
//    }

    fun isPlaying(): Boolean {
        if (mService != null) {
            return mService!!.isPlaying()
        }
        return false
    }

    fun play(uri: Uri?) {
        if (mService != null) {
            mService!!.playMusic(uri)
        }
    }

    fun nextMusic() {
        if (mService != null) {
            mService!!.nextMusic()
        }
    }

    fun prevMusic() {
        if (mService != null) {
            mService!!.prevMusic()
        }
    }

    fun clickResumeButton() {
        if (mService != null) {
            mService!!.clickResumeButton()
        }
    }

    // 사용 안함
    //    public AudioAdapter.AudioItem getAudioItem() {
    //        if (mService != null) {
    //            return mService.getAudioItem();
    //        }
    //        return null;
    //    }
//    @RequiresApi(Build.VERSION_CODES.Q)
//    fun queryAudioItem(position: Int) {
//        if (mService != null) {
//            mService!!.queryAudioItem(position)
//        }
//    }

    fun mAudioItem(): AudioItem? {
        if (mService != null) {
            return mService!!.mAudioItem
        }
        return null
    }

    init {
        mServiceConnection = object : ServiceConnection {
            override fun onServiceConnected(name: ComponentName, service: IBinder) {
                mService = (service as LocalBinder).getService()
                Log.d("Interface", "서비스 연결")
            }

            override fun onServiceDisconnected(name: ComponentName) {
                mServiceConnection = null
                mService = null
                Log.d("Interface", "서비스 연결 해제")
            }
        }
        context.bindService(Intent(context, MusicService::class.java)
                .setPackage(context.packageName), mServiceConnection as ServiceConnection, Context.BIND_AUTO_CREATE)
    }
}